execution.go

```diff
@@ -73,232 +73,165 @@ func NewEngineAPIExecutionClient(
 			return nil
 		}))
 	if err != nil {
-		ethClient.Close() // Clean up eth client if engine client fails
 		return nil, err
 	}
 
-	return &EngineAPIExecutionClient{
-		engineClient:  engineClient,
-		ethClient:     ethClient,
-		genesisHash:   genesisHash,
-		initialHeight: 1, // set to 1 and updated in InitChain
-		feeRecipient:  feeRecipient,
+	return &PureEngineClient{
+		engineClient: engineClient,
+		ethClient:    ethClient,
+		genesisHash:  genesisHash,
+		feeRecipient: feeRecipient,
 	}, nil
 }
 
-// Start starts the execution client
-func (c *EngineAPIExecutionClient) Start() error {
-	return nil
-}
-
-// Stop stops the execution client and closes all connections
-func (c *EngineAPIExecutionClient) Stop() {
-	if c.engineClient != nil {
-		c.engineClient.Close()
-	}
-
-	if c.ethClient != nil {
-		c.ethClient.Close()
-	}
-}
-
-// InitChain initializes the blockchain with genesis information
-func (c *EngineAPIExecutionClient) InitChain(ctx context.Context, genesisTime time.Time, initialHeight uint64, chainID string) (execution_types.Hash, uint64, error) {
+// InitChain initializes the blockchain with the given genesis parameters
+func (c *PureEngineClient) InitChain(ctx context.Context, genesisTime time.Time, initialHeight uint64, chainID string) ([]byte, uint64, error) {
+	// Acknowledge the genesis block
 	var forkchoiceResult engine.ForkChoiceResponse
 	err := c.engineClient.CallContext(ctx, &forkchoiceResult, "engine_forkchoiceUpdatedV3",
+		engine.ForkchoiceStateV1{
+			HeadBlockHash:      c.genesisHash,
+			SafeBlockHash:      c.genesisHash,
+			FinalizedBlockHash: c.genesisHash,
+		},
+		nil,
+	)
+	if err != nil {
+		return nil, 0, fmt.Errorf("engine_forkchoiceUpdatedV3 failed: %w", err)
+	}
+
+	// Start building the first block
+	err = c.engineClient.CallContext(ctx, &forkchoiceResult, "engine_forkchoiceUpdatedV3",
 		engine.ForkchoiceStateV1{
 			HeadBlockHash:      c.genesisHash,
 			SafeBlockHash:      c.genesisHash,
 			FinalizedBlockHash: c.genesisHash,
 		},
 		engine.PayloadAttributes{
 			Timestamp:             uint64(genesisTime.Unix()), //nolint:gosec // disable G115
-			Random:                common.Hash{},
+			Random:                common.Hash{},              // TODO(tzdybal): this probably shouldn't be 0
```

Comment:
some frameworks use blockhash as random (here it is genesis hash).

Reply:
Also part of https://github.com/rollkit/go-execution-evm/issues/34